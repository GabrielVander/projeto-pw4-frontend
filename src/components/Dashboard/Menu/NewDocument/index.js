import React from 'react';
import Card from 'react-bootstrap/Card';
import {BsFileEarmarkPlus} from 'react-icons/bs';
import './styles.css';
import { motion } from 'framer-motion';
import {useHistory} from 'react-router-dom';

function NewDocument() {
	const history = useHistory();

	return (
		<>
			<motion.div
				whileHover={{ scale: 1.1 }}
				whileTap={{ scale: 0.9 }}
			>
				<Card bg='success' onClick={() => history.push('/documents/new')}>
					<Card.Header>New document</Card.Header>
					<Card.Body>
						<BsFileEarmarkPlus size={50} />
					</Card.Body>
				</Card>
			</motion.div>
		</>
	);
}

export default NewDocument;