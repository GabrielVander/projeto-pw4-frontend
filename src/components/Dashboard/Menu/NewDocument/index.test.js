import React from 'react';
import { render } from '@testing-library/react';
import NewDocument from './index';

describe('Component renders properly', () => {
	test('Matches snapshot', () => {
		const component = render(<NewDocument/>);

		expect(component).toBeDefined();
		expect(component).toMatchSnapshot();
	});
});