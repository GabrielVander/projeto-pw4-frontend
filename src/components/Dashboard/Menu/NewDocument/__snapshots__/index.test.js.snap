// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="card bg-success"
        >
          <div
            class="card-header"
          >
            New document
          </div>
          <div
            class="card-body"
          >
            <svg
              fill="currentColor"
              height="50"
              size="50"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 1H4a2 2 0 00-2 2v10a2 2 0 002 2h5v-1H4a1 1 0 01-1-1V3a1 1 0 011-1h5v2.5A1.5 1.5 0 0010.5 6H13v2h1V6L9 1z"
              />
              <path
                clip-rule="evenodd"
                d="M13.5 10a.5.5 0 01.5.5v2a.5.5 0 01-.5.5h-2a.5.5 0 010-1H13v-1.5a.5.5 0 01.5-.5z"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M13 12.5a.5.5 0 01.5-.5h2a.5.5 0 010 1H14v1.5a.5.5 0 01-1 0v-2z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="card bg-success"
      >
        <div
          class="card-header"
        >
          New document
        </div>
        <div
          class="card-body"
        >
          <svg
            fill="currentColor"
            height="50"
            size="50"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 1H4a2 2 0 00-2 2v10a2 2 0 002 2h5v-1H4a1 1 0 01-1-1V3a1 1 0 011-1h5v2.5A1.5 1.5 0 0010.5 6H13v2h1V6L9 1z"
            />
            <path
              clip-rule="evenodd"
              d="M13.5 10a.5.5 0 01.5.5v2a.5.5 0 01-.5.5h-2a.5.5 0 010-1H13v-1.5a.5.5 0 01.5-.5z"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M13 12.5a.5.5 0 01.5-.5h2a.5.5 0 010 1H14v1.5a.5.5 0 01-1 0v-2z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
